daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1182,6 +1182,25 @@ private[lf] object SBuiltin {
       val gkey = GlobalKey(templateId, keyWithMaintainers.key)
       // check if we find it locally
       onLedger.ptx.keys.get(gkey) match {
+        case Some(Some(coid)) if onLedger.localContracts.contains(coid) =>
+          val cachedContract = onLedger.cachedContracts
+            .get(coid)
+            .getOrElse(crash(s"Local contract $coid not in cachedContracts"))
+          val stakeholders = cachedContract.signatories union cachedContract.observers
+          throw SpeedyHungry(
+            SResultNeedLocalKeyVisible(
```

Comment:
Note that this is not cached, I’ll implement caching on top of #9451 which provides the necessary infrastructure.